src/convert/streams.ts

```diff
@@ -123,54 +125,58 @@ export abstract class ComponentWriter extends Writable {
 }
 
 export class StandardWriter extends ComponentWriter {
-  public converted: SourceComponent[] = [];
+  /** filepaths that converted files were written to */
+  public readonly converted: string[] = [];
+  public readonly deleted: FileResponseSuccess[] = [];
 
-  public constructor(rootDestination: SourcePath, private resolver = new MetadataResolver()) {
+  public constructor(rootDestination: SourcePath) {
     super(rootDestination);
   }
 
   public async _write(chunk: WriterFormat, encoding: string, callback: (err?: Error) => void): Promise<void> {
     let err: Error | undefined;
     if (chunk.writeInfos.length !== 0) {
       try {
-        const toResolve: string[] = [];
+        const toResolve = new Set<string>();
         // it is a reasonable expectation that when a conversion call exits, the files of
         // every component has been written to the destination. This await ensures the microtask
         // queue is empty when that call exits and overall less memory is consumed.
         await Promise.all(
-          chunk.writeInfos.map((info: WriteInfo) => {
-            const fullDest = isAbsolute(info.output) ? info.output : join(this.rootDestination as string, info.output);
-            if (!existsSync(fullDest)) {
-              for (const ignoredPath of this.forceIgnoredPaths) {
-                if (
-                  dirname(ignoredPath).includes(dirname(fullDest)) &&
-                  basename(ignoredPath).includes(basename(fullDest))
-                ) {
+          chunk.writeInfos
+            .map(makeWriteInfoAbsolute(this.rootDestination))
+            .filter(existsOrDoesntMatchIgnored(this.forceIgnoredPaths))
+            .filter((info) => !this.forceIgnoredPaths.has(info.output))
+            .map((info) => {
+              if (info.shouldDelete) {
```

Comment:
writeInfo can now delete files.  This was an easier approach than the existing partialDelete for bundles and I think I'd like those to eventually move to this approach.

`this.deleted` accumulates the delete side effects.